servant-server/src/Servant/Server/Internal.hs

```diff
@@ -116,11 +118,10 @@ instance (KnownSymbol capture, FromText a, HasServer sublayout)
      a -> ServerT sublayout m
 
   route Proxy subserver =
-    DynamicRouter $ \ first ->
-      route (Proxy :: Proxy sublayout)
-            (case captured captureProxy first of
-               Nothing  -> return $ failWith NotFound
-               Just v   -> feedTo subserver v)
+    DynamicRouter $ \ first -> case captured captureProxy first of
+       Nothing  -> LeafRouter (\_ r -> r $ Fail err404)
+       Just v   -> route (Proxy :: Proxy sublayout) (feedTo subserver v)
+
```

Comment:
This is in deviation from the desired behaviour we discussed earlier, because it means that captures are checked immediately. In particular, this also means that if we have several routes such as
```
     "foo" :> Capture ... :> "bar"
:<|> "foo" :> Capture ... :> "baz"
```
then the `Capture` will be performed multiple times unless you left-factor by hand.